utils/patch-for-qa.sh

```diff
@@ -3,31 +3,38 @@
 CUSTOM_DATA_VM="[base64(if(empty(parameters('customData')), variables('customData'), parameters('customData')))]"
-CUSTOM_DATA_WO_HTTP_PROXY="[concat(variables('cloudConfig'), '  access_portal: ', variables('hostSetup').accessPortal, '\n  config_server: ', variables('hostSetup').configServer, '\n  notification_server: ', variables('hostSetup').notificationServer, '\n  jointoken: ', parameters('jointoken'), '\n')]"
-
+CUSTOM_DATA_WO_HTTP_PROXY="[concat(variables('cloudConfig'), '  access_portal: ', parameters('accessPortal'), '\n  config_server: ', parameters('configServer'), '\n  notification_server: ', parameters('notificationServer'), '\n  jointoken: ', parameters('jointoken'), '\n')]"
+NOTIFICATION_SERVER_PARAMETER='{"type":"string","metadata":{"description":"Notification server."},"defaultValue":""}'
 SG_SSH='{"type":"string","defaultValue":"false","allowedValues":["false","true"]}'
 SR="[if(equals(parameters('securityGroupSsh'), 'true'), concat(if(equals(parameters('securityGroupBootstrapUi'), 'true'), variables('securityRuleHtps'), json('[]')), variables('securityRuleSsh')), if(equals(parameters('securityGroupBootstrapUi'), 'true'), variables('securityRuleHtps'), json('[]')))]"
 SR_SSH='[{"name":"SSH","properties":{"description":"Allow SSH","access":"Allow","destinationAddressPrefix":"*","destinationPortRange":"22","direction":"Inbound","priority":100,"protocol":"Tcp","sourceAddressPrefix":"*","sourcePortRange":"*"}}]'
 
+# UI template variables section.
+UI_ACCESS_PORTAL_OUTPUT="[basics('hostSetup').accessPortal]"
+UI_CONFIG_SERVER_OUTPUT="[basics('hostSetup').configServer]"
 UI_CUSTOM_DATA_FIELD='[{"name":"customData","type":"Microsoft.Common.TextBox","label":"Custom data","toolTip":"Pass Custom Data for cloudinit.","multiLine":true,"constraints":{"required":false,"regex":"(?=.*).{0,100000}","validationMessage":"The Custom data cannot be more than 100000 characters."},"visible":"[equals(basics('"'"'injecting'"'"'), '"'"'Custom Data'"'"')]"}]'
 UI_CUSTOM_DATA_OUTPUT="[basics('customData')]"
+UI_HOST_SETUP_FIELD='[{"name":"hostSetup","type":"Microsoft.Common.Section","label":"Host setup","elements":[{"name":"accessPortal","type":"Microsoft.Common.TextBox","label":"Access portal","toolTip":"Access portal.","constraints":{"required":false,"regex":"(?=.*).{0,1000}","validationMessage":"The Access portal cannot be more than 1000 characters."},"visible":"true"},{"name":"configServer","type":"Microsoft.Common.TextBox","label":"Config server","toolTip":"Config server.","constraints":{"required":false,"regex":"(?=.*).{0,1000}","validationMessage":"The Config server cannot be more than 1000 characters."},"visible":"true"},{"name":"notificationServer","type":"Microsoft.Common.TextBox","label":"Notification server","toolTip":"Notification server.","constraints":{"required":false,"regex":"(?=.*).{0,1000}","validationMessage":"The Notification server cannot be more than 1000 characters."},"visible":"true"}],"visible":true}]'
```

Comment:
Could we use for validation of `accessPortal`, `configServer`, and `notificationServer` not any characters (`.*`), but regexp for the valid `URL`?
In this case the `validationMessage` will be not `The *** cannot be more than 1000 characters`, but `The *** should be a valid URL and cannot be more than 1000 characters`

Do we need this?

Reply:
Updated regex.